helios-system-tests/src/main/java/com/spotify/helios/system/SystemTestBase.java

```diff
@@ -1000,10 +1013,13 @@ public DeploymentGroupStatus call() throws Exception {
 
         if (response != null) {
           final DeploymentGroupStatus status = response.getDeploymentGroupStatus();
-          if (status.getState().equals(state)) {
+          final DeploymentGroupStatus.State actual = status.getState();
+          if (expected.equals(actual)) {
```

Comment:
* == never throws NullPointerException
* == is subject to type compatibility check at compile time
* It's faster

See more here http://stackoverflow.com/a/2937561/553994